java-tools/OsmAndServer/src/main/java/net/osmand/server/api/services/WikiService.java

```diff
@@ -58,9 +59,16 @@ public FeatureCollection getImages(String northWest, String southEast) {
 				+ " ORDER BY views desc LIMIT " + LIMIT_QUERY, "imgLat", "imgLon", null);
 	}
 	
+	public FeatureCollection getImagesById(long id) {
+		String query = String.format("SELECT id, mediaId, namespace, imageTitle, imgLat, imgLon " +
+				"FROM wikigeoimages WHERE id = %d " +
+				"ORDER BY views DESC LIMIT " + LIMIT_QUERY, id);
+		return getPoiData(null, null, query, "imgLat", "imgLon", null);
+	}
+	
 	public FeatureCollection getWikidataData(String northWest, String southEast, String lang, Set<String> filters) {
 		String filterQuery = filters.isEmpty() ? "" : "AND poitype IN (" + filters.stream().map(s -> "'" + s + "'").collect(Collectors.joining(", ")) + ")";
-		String query = "SELECT id, photoId, photoTitle, catId, catTitle, depId, depTitle, wikiTitle, wikiLang, wikiDesc, wikiArticles, osmid, osmtype, poitype, poisubtype, lat, lon "
+		String query = "SELECT * "
```

Comment:
Bad idea, we need to specify what's retrieved. So it's only very specific and transparent